bench.py

```diff
@@ -0,0 +1,416 @@
+
+    return output
+
+# Parse the output file - separete output lines for in and out-of place.
+# For each matching line in the file return size and time for out-of-place and in-place as separate lines
+# Code is based on based on https://github.com/microsoft/msccl-leaderboard/blob/main/generate_graphs.py#L48
+def parse_test_log_inout_separate(path):
+    # Example input
+    # [1,0]<stdout>:#
+    # [1,0]<stdout>:#                                                              out-of-place                       in-place
+    # [1,0]<stdout>:#       size         count      type   redop    root     time   algbw   busbw #wrong     time   algbw   busbw #wrong
+    # [1,0]<stdout>:#        (B)    (elements)                               (us)  (GB/s)  (GB/s)            (us)  (GB/s)  (GB/s)
+    # [1,0]<stdout>:az-eus-v100-32gb-5-worker-zphjiy:18075:18075 [0] NCCL INFO Launch mode Parallel
+    # [1,0]<stdout>:        1024           256     float     sum      -1    75.18    0.01    0.03      0    78.30    0.01    0.02      0
+    # [1,0]<stdout>:        2048           512     float     sum      -1    79.54    0.03    0.05      0    82.56    0.02    0.05      0
+
+    # output is an array of string
+    output = []
+
+    # Create a pattern for lines that begin with '[1,0]<stdout>: ' and contain a number followed by a space and then a number
+    pattern = re.compile(f'(\[1,0\]<stdout>:)?\s*(\d+)\s+(\d+)\s+')
+    with open(path) as f:
+        for line in f.readlines():
+            m = pattern.match(line)
+            if m is not None:
+                # split the line into an array of strings were each string is separated by one or more spaces
+                lineArray = line.split()
+                # Create a common separated string with the size, first time columns
+                output.append(f"outPlace,{lineArray[1]},{lineArray[6]}")
+
+                # Create a common separate string with the size and the second time columns
+                output.append(f"inPlace,{lineArray[1]},{lineArray[10]}")
+
+    return output
+
+# Format the header row for the CSV output file
+# param: separator is the character used to separate the columns in the CSV file
+# Returns the string for the header row
+def format_header_row(separator) -> str:
+    header = ('Collective'
+        + separator + 'Filter'
+        + separator + 'TimeOfTest'
+        + separator + 'nGPUs'
+        + separator + 'Protocol'
+        + separator + 'Library'
+        + separator + 'NcclVersion'
+        + separator + 'Size'
+        + separator + 'Time_OutOfPlace'
+        + separator + 'Time_InPlace'
+        + separator + 'MSCCL_Algo_Name'
+        + separator + 'MSCCL_File'
+        + '\n')
+    return header
+
+# This is and example of a MSCCL input file
+# Based on sccl-presynth/sccl_presynth/Allgather.n16-DistributedRelayedSwitch.localDGX1.copies2-steps20-gurobisol-improve-1630536923_i8_scRemote1_IBContig_h5-noring.sccl.xml
+exampleMCCLInput = '''
```

Comment:
For testing purposes, I could commit a directory fake MSCCL XML files and have a test that runs bench.py to iterate over them.  I'll consider this after collecting more feedback.